src/world/mod.rs

```diff
@@ -34,6 +34,20 @@ impl<'a> Iterator for CreateIter<'a> {
     }
 }
 
+/// A common trait for EntityBuilder and LazyBuilder, allowing either to be used.
+pub trait Builder {
```

Comment:
Might want to point out the guarantees here:

* entity is definitely alive
* components may or may not exist before a `maintain`